crawler/index.html

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta encoding='utf-8'/>
+    <title>Headless detector</title>
+  </head>
+
+  <body>
+    <table>
+        <tr>
+            <th>Test</th>
+            <th>Result</th>
+        </tr>
+        <tbody id="results"></tbody>
+    </table>
+    <template id="result-row">
+        <tr>
+            <td class="key"></td>
+            <td class="value"></td>
+        </tr>
+    </template>
+    <script>
+        (async () => {
+            const ul = document.getElementById('results');
+            const template = document.getElementById('result-row');
+            
+            function displayResult(test, result, detectValue) {
+                const listEntry = template.content.cloneNode(true);
+                listEntry.querySelector('.key').textContent = test;
+                listEntry.querySelector('.value').textContent = result ? 'fail' : 'pass';
+                ul.appendChild(listEntry)
+                if (result) {
+                    fetch(`./fail.json?test=${test}&${detectValue}`);
+                }
+            }
+
+            displayResult('userAgent', /HeadlessChrome/.test(window.navigator.userAgent), window.navigator.userAgent);
+            displayResult('plugins', navigator.plugins.length == 0, navigator.plugins.length);
+            displayResult('languages', navigator.languages == "", navigator.languages);
+            displayResult('webdriver', !!navigator.webdriver, navigator.webdriver)
+            displayResult('window.chrome', !window.chrome || !window.chrome.runtime, JSON.stringify(window.chrome))
```

Comment:
FYI there is no `window.chrome.runtime` in Chrome Canary (90) and my plugins are in fact empty. Probably not a big deal, but shows that those tests are somehow not reliable.

Reply:
The articles around this also stress that some of these methods only work with certain versions. Having this also as a webpage means we can also test against normal chrome, and remove tests which no longer pass for real browsers.